pkg/controller.v1alpha3/experiment/experiment_util.go

```diff
@@ -109,3 +111,31 @@ func (r *ReconcileExperiment) updateFinalizers(instance *experimentsv1alpha3.Exp
 		return reconcile.Result{Requeue: true}, err
 	}
 }
+
+func (r *ReconcileExperiment) terminateSuggestion(instance *experimentsv1alpha3.Experiment) (reconcile.Result, error) {
+	log.Info("Start terminating original...")
+	original := &suggestionsv1alpha3.Suggestion{}
+	err := r.Get(context.TODO(), types.NamespacedName{
+		Namespace: instance.Namespace,
+		Name:      instance.Name,
+	}, original)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, err
+	}
+	if original.IsCompleted() {
+		return reconcile.Result{}, nil
+	}
+	suggestion := original.DeepCopy()
+	msg := "Suggestion is succeeded"
+	suggestion.MarkSuggestionStatusSucceeded(suggestionController.SuggestionSucceededReason, msg)
+	log.Info("Mark suggestion succeeded...")
+
+	if err := r.Update(context.TODO(), suggestion); err != nil {
```

Comment:
@sperlingxx  Let's keep `UpdateSuggestion(suggestion, 0)` call here to be consistent with https://github.com/kubeflow/katib/blob/master/pkg/controller.v1alpha3/experiment/experiment_controller.go#L391. I will modify this function to not receive `suggestionRequests`.
To fix e2e test, did you run mock generator: https://github.com/kubeflow/katib/blob/master/scripts/mockgen.sh?

Reply:
It's done.